src/components/ui/ImageGallery/ImageGallerySelector.tsx

```diff
@@ -0,0 +1,128 @@
+import type { HTMLAttributes, PropsWithChildren } from 'react'
+import React, { useMemo } from 'react'
+import { useSlider, IconButton } from '@faststore/ui'
+
+import { ForwardArrowIcon, BackwardArrowIcon } from './Icons'
+
+interface Props extends HTMLAttributes<HTMLDivElement> {
+  itemsPerPage: number
+}
+
+const createTransformValues = (totalItems: number) => {
+  if (!totalItems) {
+    return {}
+  }
+
+  const transformMap: Record<number, number> = {}
+  const slideWidth = 100 / totalItems
+
+  for (let idx = 0; idx < totalItems; ++idx) {
+    const transformValue = -(slideWidth * idx)
+
+    transformMap[idx] = transformValue
+  }
+
+  return transformMap
+}
+
+function ImageGallerySelector({
+  itemsPerPage,
+  children,
+  ...otherProps
+}: PropsWithChildren<Props>) {
+  const elements = React.Children.toArray(children)
+  const elementCount = elements.length
+
+  const transformValues = useMemo(
+    () => createTransformValues(elementCount),
+    [elementCount]
+  )
+
+  const { handlers, slide, sliderState, sliderDispatch } = useSlider({
+    totalItems: elementCount,
+    itemsPerPage,
+    infiniteMode: false,
+  })
+
+  if (!elementCount) {
+    return null
+  }
+
+  return (
+    <section aria-roledescription="carousel" aria-label="Product images">
+      <IconButton
+        aria-label="slide image selector backward"
+        icon={<BackwardArrowIcon color="#323845" />}
+        onClick={() => {
+          if (sliderState.sliding) {
+            return
+          }
+
+          slide('previous', sliderDispatch)
+        }}
+      />
+      <IconButton
+        aria-label="slide image selector forward"
```

Comment:
What do you think about this suggestion?
I'm not 100% sure about the English, but for me putting it at the beginning sounds better and starts with the most important part of the component description
```suggestion
        aria-label="backward slide image selector"
        icon={<BackwardArrowIcon color="#323845" />}
        onClick={() => {
          if (sliderState.sliding) {
            return
          }

          slide('previous', sliderDispatch)
        }}
      />
      <IconButton
        aria-label="forward slide image selector"
```

Reply:
Great suggestion. I believe the English is also fine.